src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -235,7 +235,12 @@ def parse(self, backend, x509_obj):
                     )
             else:
                 d2i = backend._lib.X509V3_EXT_d2i(ext)
-                assert d2i != backend._ffi.NULL
+                if d2i == backend._ffi.NULL:
+                    backend._consume_errors()
+                    raise ValueError(
+                        "The {0} extension appears to be corrupt".format(oid)
```

Comment:
Corrupt seems like an unnescsary normative judgement. Why not "is invalid and cannot be parsed"?